### ISSUE TITLE
Please restructure for proper build and packaging

### ISSUE THREAD

**Issue by techzilla** [Isma399/check_4_icinga#1]
Please restructure for proper build and packaging, these plugins would be worth using, but I'm not even able to unless they are packaged. 

First the build needs to be setup a somwhat standard convention, and then we could easily build from spec or make a release package. 

Is CMAKE something you'd consider?

**Comment by Isma399** [comment]
Hello,

have you tried to compele with gcc: 
`gcc net-snmp-config --cflags net-snmp-config --libs net-snmp-config --external-libs check_linux_xxx.c -o `
I give a try to CMAKE but I didn't manage to link SNMP librairires.
```
[root@nagios testCMAKE]# cat CMakeLists.txt
cmake_minimum_required(VERSION 2.6)
project(Check_linux_disk)
include(FindNetSnmp.cmake)
find_library(NetSnmp REQUIRED)
add_executable(Check_linux_disk  check_linux_disk.c)
```
[FindNetSNmp.cmake](https://gist.githubusercontent.com/michaelmelanson/111912/raw/3bb504820662cd753b75db9a31d37a5ba840f444/FindNetSNMP.cmake)


```

**Comment by techzilla** [comment]
it did build when doing the command manually, I also had some pretty serious problems when trying to CMake it.   However getting this in a proper convention will enable us to get this packaged easier, I had to do it all very manually to deploy them. 

Once we get that far, we can also get in the package Icinga2 checkcommand config files, I already created some we could start from. So far they seem to be working, but I havn't messed with the thresholds yet. 

I'll see if I can get CMake working, will work on this for a bit, and report back findings.
